src/netaddress.cpp

```diff
@@ -1089,28 +1100,53 @@ bool CSubNet::Match(const CNetAddr &addr) const
     if (!valid || !addr.IsValid() || network.m_net != addr.m_net)
         return false;
     assert(network.m_addr.size() == addr.m_addr.size());
-    for (size_t x = 0; x < addr.m_addr.size(); ++x) {
-        if ((addr.m_addr[x] & netmask[x]) != network.m_addr[x]) {
-            return false;
+    switch (network.m_net) {
+    case NET_IPV4:
+    case NET_IPV6:
+        for (size_t x = 0; x < addr.m_addr.size(); ++x) {
+            if ((addr.m_addr[x] & netmask[x]) != network.m_addr[x]) {
+                return false;
+            }
         }
+        return true;
+    case NET_ONION:
+    case NET_I2P:
+    case NET_CJDNS:
+        return addr == network;
+    case NET_INTERNAL:
+    case NET_UNROUTABLE:
+    case NET_MAX:
+        return false;
     }
-    return true;
+    return false;
 }
 
 std::string CSubNet::ToString() const
 {
-    assert(network.m_addr.size() <= sizeof(netmask));
+    switch (network.m_net) {
+    case NET_IPV4:
+    case NET_IPV6: {
+        assert(network.m_addr.size() <= sizeof(netmask));
 
-    uint8_t cidr = 0;
+        uint8_t cidr = 0;
 
-    for (size_t i = 0; i < network.m_addr.size(); ++i) {
-        if (netmask[i] == 0x00) {
-            break;
+        for (size_t i = 0; i < network.m_addr.size(); ++i) {
+            if (netmask[i] == 0x00) {
+                break;
+            }
+            cidr += NetmaskBits(netmask[i]);
         }
-        cidr += NetmaskBits(netmask[i]);
-    }
 
-    return network.ToString() + strprintf("/%u", cidr);
+        return network.ToString() + strprintf("/%u", cidr);
+    }
+    case NET_ONION:
+    case NET_I2P:
+    case NET_CJDNS:
+    case NET_INTERNAL:
+    case NET_UNROUTABLE:
+    case NET_MAX:
+        return network.ToString();
+    }
```

Comment:
Control reaches end of non-void function.

Reply:
Fixed.